### ISSUE TITLE
Bulk Edit(Select) Mode

### ISSUE THREAD

**Issue by girishghoda** [MessageKit/MessageKit#447]
## General Information

* MessageKit Version:

* iOS Version(s):

* Swift Version:

* Devices/Simulators:

* Reproducible in ChatExample? (Yes/No):

## What did you expect to happen?
@SD10 Here I want to new feature (I don't know its already present or not)
Feature is Bulk Edit mode like select multiple messages like 'JSQMessagesViewController' bulk edit mode
https://github.com/jessesquires/JSQMessagesViewController/pull/1593

Actually i want this functionality for selected single/multiple messages for forward/delete
Thank you

**Comment by girishghoda** [comment]
Hey @BasThomas for now i want to achieve this functionality,if any alternative solution in Message kit?
Any suggestion appriciated
Thank you

**Comment by SD10** [comment]
@girishghoda Yes this is a feature request, it does not exist currently. I’m not sure when we’ll support this

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by girishghoda** [comment]
@BasThomas @SD10 I have hack(suggestion) for everyone if you block on this feature
**Its for Only one message at time**

**1)Add method in MessageCellDelegate protocol**
 func didTapLongPressed(in cell: MessageCollectionViewCell)

**2)Add Long pressed gesture in MessageCollectionViewCell**
```
//Long press cellview
    func setupLongPressGestureRecognizers() {
        let longPressGesture:UILongPressGestureRecognizer = 
        UILongPressGestureRecognizer(target: self, action: #selector(handleLongPress))
        longPressGesture.minimumPressDuration = 1.0 // 1 second press
        //longPressGesture.delegate = self
        self.addGestureRecognizer(longPressGesture)
    }
```
    /// Handle cell long press
    @objc func handleLongPress(_ longPressGestureRecognizer: UILongPressGestureRecognizer) {
        
        if longPressGestureRecognizer.state == UIGestureRecognizerState.began {
            
            delegate?.didTapLongPressed(in: self)
        }
    }

**3)Implement method in your Chat view :** 
  func didTapLongPressed(in cell: MessageCollectionViewCell) {
  }
Its my hack for temporary work on single message delete/forward, and i add here solution for everyone

Thank you

**Comment by girishghoda** [comment]
` //Long press cellview
    func setupLongPressGestureRecognizers() {
        let longPressGesture:UILongPressGestureRecognizer = UILongPressGestureRecognizer(target: self, action: #selector(handleLongPress))
        longPressGesture.minimumPressDuration = 1.0 // 1 second press
        //longPressGesture.delegate = self
        self.addGestureRecognizer(longPressGesture)
    }
`

**Comment by AndreiBoariu** [comment]
@SD10 Any update on this ? Do you have any idea if someone started working on this ?
Appreciate

**Comment by lhr000lhr** [comment]
Any update on this? Does anybody have any solution?
How to build the interface like this?
![IMG_7BA206152B94-1](https://user-images.githubusercontent.com/8776111/81517113-efbc7a80-938d-11ea-84e7-d61a6384b675.jpeg)

**Comment by balajiechat** [comment]
Hey, Any update on the above feature. Please let me know how can we achieve this?

**Comment by lhr000lhrmega** [comment]
Hi @balajiechat 

Yes, we can.
I have already implemented Edit feature using a custom `Supplementary View`, pretty simple actually.
<img width="561" alt="Screen Shot 2020-08-18 at 9 35 34 AM" src="https://user-images.githubusercontent.com/56808021/90446839-31e81b80-e136-11ea-994f-940ec09c8bfa.png">

**Comment by warisappify** [comment]
@lhr000lhrmega then why don't you help others and give the example code here.

**Comment by lhr000lhrmega** [comment]
> @lhr000lhrmega then why don't you help others and give the example code here.

https://github.com/meganz/iOS/blob/master/iMEGA/Chat%20Messages/ChatViewMessagesFlowLayout.swift#L72

**Comment by reborncho** [comment]
https://github.com/meganz/iOS/blob/master/iMEGA/Chat%20Messages/ChatViewMessagesFlowLayout.swift#L72

Does this code implement multi-select functionality?
